src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -127,27 +255,66 @@ public override void LayoutSubviews()
 				return;
 			}
 
-			var bounds = AdjustForSafeArea(Bounds).ToRectangle();
+			if (!ValidateSafeArea())
+			{
+				InvalidateConstraintsCache();
+
+				// The safe area is now interacting differently with our view, so we have to enqueue a second layout pass
+				// to let ancestors adjust to the measured size.
+				if (this.IsFinalMeasureHandledBySuperView())
+				{
+					SetNeedsLayout();
+					this.InvalidateAncestorsMeasures();
+					return;
+				}
+			}
+
+			var bounds = Bounds.ToRectangle();
 
 			var widthConstraint = bounds.Width;
 			var heightConstraint = bounds.Height;
 
-			// If the SuperView is a cross-platform layout backed view (i.e. MauiView, MauiScrollView, LayoutView, ..),
-			// then measurement has already happened via SizeThatFits and doesn't need to be repeated in LayoutSubviews.
-			// This is especially important to avoid overriding potentially infinite measurement constraints
-			// imposed by the parent (i.e. scroll view) with the current bounds, except when our bounds are fixed by constraints.
-			// But we _do_ need LayoutSubviews to make a measurement pass if the parent is something else (for example,
-			// the window); there's no guarantee that SizeThatFits has been called in that case.
-			if (!IsMeasureValid(widthConstraint, heightConstraint) && !this.IsFinalMeasureHandledBySuperView() ||
-				!HasBeenMeasured() && HasFixedConstraints)
+			// TODO: This is expensive and should not be happening during arrange
+			// This happens on the first layout pass when the default size of the page is set
+			// If we're here and the fixed constraints have changed, it's likely that the content was invalidated 
+			// and we need to remeasure
+			if (!IsMeasureValid(widthConstraint, heightConstraint))
 			{
-				CrossPlatformMeasure(widthConstraint, heightConstraint);
+				var crossPlatformSize = CrossPlatformMeasure(widthConstraint, heightConstraint);
 				CacheMeasureConstraints(widthConstraint, heightConstraint);
 			}
 
 			CrossPlatformArrange(bounds);
 		}
 
+		bool ValidateSafeArea()
+		{
+			// If nothing changed, we don't need to do anything
+			if (!_safeAreaInvalidated)
+			{
+				return true;
+			}
+
+			// Mark the safe area as validated given that we're about to check it
+			_safeAreaInvalidated = false;
+
+			// Store the information about the safe area for developers to use
+			if (View is ISafeAreaPage safeAreaPage)
+			{
+				safeAreaPage.SafeAreaInsets = SafeAreaInsets.ToThickness();
+			}
+
+			var oldSafeArea = _safeArea;
+			_safeArea = SafeAreaInsets.ToSafeAreaInsets();
```

Comment:
This code needs to be enhanced to take into account the new flags.

If the software keyboard is present this value needs to be a rectangle representing the software keyboard size.

That way the view gets inset enough and none of it is behind the keyboard